ice/settings.py

```diff
@@ -27,10 +32,45 @@ class Settings(BaseSettings):
     OUGHT_ICE_AUTO_BROWSER: bool = True
     PAPER_DIR: Path = Path(__file__).parent.parent / "papers"
 
+    # note these attributes are read differently- see [__getattribute__]
+    OPENAI_API_KEY: str = ""
+    OUGHT_INFERENCE_API_KEY: str = ""
+    ELICIT_AUTH_TOKEN: str = ""
+
+    def __get_and_store(self, setting_name: str, prompt: Optional[str] = None) -> str:
+        # We use [__getattribute__] to read these attributes, so that we can
+        # prompt the user for them if they are not already set.
+        if prompt is None:
+            prompt = f"Enter {setting_name}: "
+        if self.__dict__[setting_name] == "":
+            # TODO someday: one way this stinks is that starting the server will
+            # bury this prompt. (the server is started in a background process
+            # and has some log messages). i don't know if there's a clean way
+            # to fix this without re-architecting some stuff.
+            value = input(prompt)
+            setattr(self, setting_name, value)
+            with open(_env_path, "a") as f:
+                # [json.dumps] to escape quotes
+                f.write(f"{setting_name}={json.dumps(value)}\n")
+        return self.__dict__[setting_name]
 
-OUGHT_ICE_DIR = Path(environ.get("OUGHT_ICE_DIR", Path.home() / ".ought-ice"))
+    def __getattribute__(self, __name: str) -> Any:
+        match __name:
+            case "OPENAI_API_KEY":
+                return self.__get_and_store(
+                    "OPENAI_API_KEY",
+                    "Enter OpenAI API key (you can get this from https://beta.openai.com/account/api-keys): ",
+                )
+            case "OUGHT_INFERENCE_API_KEY":
+                return self.__get_and_store("OUGHT_INFERENCE_API_KEY")
+            case "ELICIT_AUTH_TOKEN":
+                return self.__get_and_store("ELICIT_AUTH_TOKEN")
+            case _:
+                return super().__getattribute__(__name)
```

Comment:
Optional: I think if you move this line to the top of this method, check if it's an empty string there, and do some other rearrangement, you can make things a bit cleaner, e.g. only have one call to `__get_and_store`, not repeat the setting name, and avoid both uses of `self.__dict__[setting_name]`.

Reply:
> check if it's an empty string there

Maybe I'm confused: doesn't the correctness of this check for emptiness depend on the value of `name`? So then we'd still need to do some casing on its value. 

> repeat the setting name

Yeah, I agree this is gross. One refactor to handle it (and your comment re: one call to `__get_and_store`) is to divide attributes into those that require this logic and those that don't. (E.g. with a `Set[str])`. Seems a bit extra, perhaps.

Reply:
Wdyt?

Reply:
Discussed over slack and worked on in 66579eacb714e201e73be5af09e4674ae709e3d5 :)